File: solutions/implement-strstr/implement-strstr.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */

// check edge case of empty needle input
    // return 0 if empty
// return indexOf needle from haystack

var strStr = function(haystack, needle) {
    if (!needle.length) {
        return 0;
    }
    
    return haystack.indexOf(needle);
};

// Time complexity = O(N) for n items in haystack using indexOf method.
// Space complexity = O(1) 

File: solutions/implement-strstr/implement-strstr.py
# check edge case of empty needle input
#     return 0 if empty
# return -1 if needle str is not in haystack str otherwise return its index using index method

class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if not len(needle):
            return 0
        
        return -1 if needle not in haystack else haystack.index(needle)
        
# Time complexity = O(N) for n items in haystack using index method.
# Space complexity = O(1) 
